core/templates/pages/feedback-updates-page/feedback-updates-page.component.spec.ts

```diff
@@ -0,0 +1,848 @@
+      const focusSpy = spyOn(focusManagerService, 'setFocusWithoutScroll');
+      spyOn(userService, 'getUserInfoAsync').and
+        .callFake(async() => {
+          return Promise.resolve(userInfo as UserInfo);
+        });
+
+      component.ngOnInit();
+      flush();
+
+      expect(focusSpy).toHaveBeenCalledWith('ourLessonsBtn');
+    }));
+
+    it('should subscribe to onLangChange upon initialisation and set page ' +
+    'title whenever language changes', fakeAsync(() => {
+      spyOn(component.directiveSubscriptions, 'add');
+      spyOn(translateService.onLangChange, 'subscribe');
+      spyOn(component, 'setPageTitle');
+
+      component.ngOnInit();
+      flush();
+
+      expect(component.directiveSubscriptions.add).toHaveBeenCalled();
+      expect(translateService.onLangChange.subscribe).toHaveBeenCalled();
+
+      translateService.onLangChange.emit();
+
+      expect(component.setPageTitle).toHaveBeenCalled();
+    }));
+
+    it('should obtain translated page title and set it', () => {
+      spyOn(translateService, 'instant').and.callThrough();
+      spyOn(pageTitleService, 'setDocumentTitle');
+
+      component.setPageTitle();
+
+      expect(translateService.instant).toHaveBeenCalledWith(
+        'I18N_FEEDBACK_UPDATES_PAGE_TITLE');
+      expect(pageTitleService.setDocumentTitle).toHaveBeenCalledWith(
+        'I18N_FEEDBACK_UPDATES_PAGE_TITLE');
+    });
+
+    it('should get static image url', () => {
+      let imagePath = '/path/to/image.png';
+
+      expect(component.getStaticImageUrl(imagePath)).toBe(
+        '/assets/images/path/to/image.png');
+    });
+
+    it('should get user profile image png data url correctly', () => {
+      expect(component.getauthorPicturePngDataUrl('username')).toBe(
+        'profile-image-url-png');
+    });
+
+    it('should get user profile image webp data url correctly', () => {
+      expect(component.getauthorPictureWebpDataUrl('username')).toBe(
+        'profile-image-url-webp');
+    });
+
+    it('should show username popover based on its length', () => {
+      expect(component.showUsernamePopover('abcdefghijk')).toBe('mouseenter');
+      expect(component.showUsernamePopover('abc')).toBe('none');
+    });
+
+    it('should change feedback sorting options by last update msecs when' +
+      ' changing sorting type', () => {
+      expect(component.isCurrentFeedbackSortDescending).toBe(true);
+      expect(component.currentFeedbackThreadsSortType).toBe('lastUpdatedMsecs');
+
+      component.setFeedbackSortingOptions('lastUpdatedMsecs');
+
+      expect(component.isCurrentFeedbackSortDescending).toBe(false);
+    });
+
+    it('should change feedback sorting options by exploration when changing' +
+      ' sorting type', () => {
+      component.setFeedbackSortingOptions('exploration');
+
+      expect(component.currentFeedbackThreadsSortType).toBe('exploration');
+      expect(component.isCurrentFeedbackSortDescending).toBe(true);
```

Comment:
Please add a pre-check

Reply:
Already written Please see here(https://github.com/oppia/oppia/blob/develop/core/templates/pages/learner-dashboard-page/learner-dashboard-page.component.spec.ts#L677)

Reply:
We need to add a a pre-check there too.